be/src/runtime/memory/thread_mem_tracker_mgr.h

```diff
@@ -209,14 +236,58 @@ inline void ThreadMemTrackerMgr::flush_untracked_mem() {
     _stop_consume = true;
     DCHECK(_limiter_tracker_raw);
 
-    old_untracked_mem = _untracked_mem;
-    if (_count_scope_mem) _scope_mem += _untracked_mem;
-    _limiter_tracker_raw->consume(old_untracked_mem);
-    for (auto tracker : _consumer_tracker_stack) {
-        tracker->consume(old_untracked_mem);
+    _old_untracked_mem = _untracked_mem;
+    if (_count_scope_mem) {
+        _scope_mem += _untracked_mem;
+    }
+    _limiter_tracker_raw->consume(_old_untracked_mem);
+    for (auto* tracker : _consumer_tracker_stack) {
+        tracker->consume(_old_untracked_mem);
     }
-    _untracked_mem -= old_untracked_mem;
+    _untracked_mem -= _old_untracked_mem;
     _stop_consume = false;
 }
 
+inline bool ThreadMemTrackerMgr::try_reserve(int64_t size, bool force_tracker_overcommit) {
```

Comment:
why need overcommit?

Reply:
default query/load allows overcommit, if `force_tracker_overcommit` is false, try_consume will not check the mem_limit of MemTracker.

Reply:
delete `force_tracker_overcommit`